### PR TITLE
Update sys cargo

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Generated `Cargo.toml` now looks like this (for sourceview-sys):

```toml
[package]
name = "gtk-source-sys"
version = "0.0.1"
links = "gtk_source"
build = "build.rs"

[lib]
name = "gtk_source_sys"

[dependencies]
libc = "0.2"

[dependencies.cairo-sys]
git = "https://github.com/gtk-rs/sys"

[dependencies.gdk-sys]
git = "https://github.com/gtk-rs/sys"

[dependencies.gdk_pixbuf-sys]
git = "https://github.com/gtk-rs/sys"

[dependencies.gio-sys]
git = "https://github.com/gtk-rs/sys"

[dependencies.glib-sys]
git = "https://github.com/gtk-rs/sys"

[dependencies.gobject-sys]
git = "https://github.com/gtk-rs/sys"

[dependencies.gtk-sys]
git = "https://github.com/gtk-rs/sys"

[build-dependencies]
pkg-config = "0.3.7"

[dev-dependencies]
shell-words = "0.1.0"
tempdir = "0.3"

[features]
v2_0 = []
v2_2 = ["v2_0"]
v2_4 = ["v2_2"]
v2_8 = ["v2_4"]
v2_10 = ["v2_8"]
v2_12 = ["v2_10"]
v3_0 = ["v2_12"]
v3_4 = ["v3_0"]
v3_8 = ["v3_4"]
v3_10 = ["v3_8"]
v3_12 = ["v3_10"]
v3_14 = ["v3_12"]
v3_16 = ["v3_14"]
v3_18 = ["v3_16"]
v3_20 = ["v3_18"]
v3_22 = ["v3_20"]
v3_24 = ["v3_22"]
dox = []
```

cc @EPashkin @sdroege

### CODE REVIEW

**Comment by sdroege** [comment]
What exactly does it change?

**Comment by GuillaumeGomez** [comment]
It forces to keep the order of the different toml sections (used in our case to generate the `Cargo.toml` file) and improve a bit the different dependencies.

**Comment by sdroege** [comment]
Sounds good

**Comment by EPashkin** [comment]
IMHO in `fill_empty` fixing package names unneeded as it called only for new packages,
it better done manually when needed.
better only leave `version` change

File: src/codegen/sys/cargo_toml.rs [hunk 1]
```diff
@@ -55,10 +54,21 @@ fn fill_empty(root: &mut Table, env: &Env, crate_name: &str) {
...
     let deps = upsert_table(root, "dependencies");
     for ext_lib in &env.config.external_libraries {
-        let ext_package = format!("{}-sys", ext_lib.crate_name);
+        let ext_package = if ext_lib.crate_name == "cairo" {
+            format!("{}-sys-rs", ext_lib.crate_name)
+        } else if ext_lib.crate_name == "gdk_pixbuf" {
+            format!("gdk-pixbuf-sys")
+        } else {
+            format!("{}-sys", ext_lib.crate_name)
+        };
```
Comment (EPashkin) [inline src/codegen/sys/cargo_toml.rs]: @GuillaumeGomez You still want this part and next?

Comment (GuillaumeGomez) [reply src/codegen/sys/cargo_toml.rs]: Yes. You see an issue with them?

Comment (EPashkin) [reply src/codegen/sys/cargo_toml.rs]: This function called only for new crates,
it too linked with gtk-rs,
and IMHO will be changed manually in most cases.
So it not bad but just unneeded.

Comment (GuillaumeGomez) [reply src/codegen/sys/cargo_toml.rs]: Those crates are specific to gtk-rs anyway. I prefer having them generated nicely instead of the current output.

Comment (EPashkin) [reply src/codegen/sys/cargo_toml.rs]: Then I merge

Comment (GuillaumeGomez) [reply src/codegen/sys/cargo_toml.rs]: Thanks! :)

**Pull request merged**